pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -477,20 +477,24 @@ func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(clusterPort int, bind
 				},
 			},
 		}
-		// add extra source addresses to cluster builder
-		var extraSrcAddrs []*core.ExtraSourceAddress
-		for _, extraBdIP := range cb.passThroughBindIPs[1:] {
-			extraSrcAddr := &core.ExtraSourceAddress{
-				Address: &core.SocketAddress{
-					Address: extraBdIP,
-					PortSpecifier: &core.SocketAddress_PortValue{
-						PortValue: uint32(0),
+		instExtraSvcAddr := instance.Service.GetExtraAddressesForProxy(proxy)
+		// the extra source address for UpstreamBindConfig shoulde be added when the service is a dual stack k8s service
+		if features.EnableDualStack && len(cb.passThroughBindIPs) > 1 && len(instExtraSvcAddr) > 0 {
```

Comment:
I donot understand why check  `instExtraSvcAddr`.

AddressMap.Addresses may hold multi addresses for Loadbalancer, does it have relationship with Dual stack. Why

Reply:
Hi @hzxuzhonghu, There are 2 method for service: `GetAddressForProxy` and `GetExtraAddressesForProxy`, the former returns the first address of service and the latter returns the rest of service. And their returned values are from the `AddressMap.Addresses`.

For the checking `instExtraSvcAddr`: the service may be deployed in a dual stack environment, but it can be deployed with single stack. In such situation,  we should not create the configuration of  `upstream_bind_config` with `extra_source_addresses`.

Reply:
In a single stack, why would passThroughBindIPs len >1?
So looks can use isDualStackCluster to replace these conditions

Reply:
In fact, it is lead by K8S,  even a service deployed with single stack mode in a dual stack environment, it means that the service has only one single IP, but it's endpoints still have 2 IPs. And `isDualStackCluster` will check the endpoint level, but not service level

Reply:
kind of understand what you mean. Feels very confused